include/dali/core/geom/box.h

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_GEOM_BOX_H_
+#define DALI_CORE_GEOM_BOX_H_
+
+#include "dali/core/geom/vec.h"
+
+namespace dali {
+
+template<size_t ndims, typename CoordinateType>
+struct Box {
+  static_assert(std::is_standard_layout<CoordinateType>::value,
+                "Cannot create a Box of a non-standard layout type");
+  using corner_t = vec<ndims, CoordinateType>;
+  static_assert(std::is_pod<corner_t>::value, "Corner has to be POD");
+
+  /**
+   * Corners of the box.
+   * Assumes, that `lo <= hi`, i.e. every coordinate of `lo` will be lower or equal to
+   * corresponding coordinate of `hi`.
+   */
+  corner_t lo, hi;
+
+
+  /**
+   * Creates an empty box
```

Comment:
That is _NOT_ true. This leaves the box uninitialized - and that's OK, just remove the comment.
Nitpick: having _three_ lines of code for a _one_ line comment seems rather wasteful.